lib/templ.ml

```diff
@@ -210,19 +268,24 @@ and eval_simple_variable conf =
   | "suffix" ->
     (* On supprime de env toutes les paires qui sont dans (henv @ senv) *)
     let l = List.fold_left (fun accu (k, _) -> List.remove_assoc k accu) conf.env (List.rev_append conf.henv conf.senv) in
-    List.fold_left (fun c (k, v) ->
-      let v = Adef.as_string v in
-      if ( ( k = "oc" || k = "ocz" ) && v = "0" ) || k = "" then c
-      else c ^ k ^ "=" ^ v ^ "&")
-    "" l
+    let s = List.fold_left (fun c (k, v) ->
+        let v = Adef.as_string v in
+        if ( ( k = "oc" || k = "ocz" ) && v = "0" ) || k = "" then c
+        else c ^ k ^ "=" ^ v ^ "&")
+      "" l
+    in
+    if String.length s >= 1 && s.[String.length s - 1] = '&' then String.sub s 0 (String.length s - 1)
+    else s (* suppress last & *)
```

Comment:
Same here, instead of building the string in a fold_left I would do in 2 steps to make it clear and remove the need to remove the last '&':
```
 let l = List.map (fun (k, v) ->
      let v = Adef.as_string v in
      if ( ( k = "oc" || k = "ocz" ) && v = "0" ) || k = "" then ""
      else Format.sprintf "%s=%s" k v) in
    String.concat "&" l
```